static/app/components/events/eventStatisticalDetector/eventDifferentialFlamegraph.tsx

```diff
@@ -316,6 +369,267 @@ function DifferentialFlamegraphTransactionToolbar(
   );
 }
 
+interface PaginationReducerState {
+  page: number;
+  pageCount: number;
+  pageSize: number;
+}
+
+type PaginationReducerAction =
+  | {type: 'next'}
+  | {type: 'previous'}
+  | {list: any[]; pageSize: number; type: 'initialize'};
+
+function paginationReducer(
+  state: PaginationReducerState,
+  action: PaginationReducerAction
+): PaginationReducerState {
+  switch (action.type) {
+    case 'initialize': {
+      return {
+        page: 0,
+        pageCount: Math.ceil(action.list.length / action.pageSize),
+        pageSize: action.pageSize,
+      };
+    }
+    case 'next':
+      return {
+        ...state,
+        page: Math.min(state.page + 1, state.pageCount),
```

Comment:
Is there an off by one here? Should it be `state.pageCount - 1` since the page numbers go from `0` to `state.pageCount - 1`?

Reply:
The numbers for pageCount are 0 - pageCount inclusive as we Math.ceil on the list length / size

Reply:
If we take 21 items with a page size of 10, then we have pages 0, 1, and 2 right? but ceiling of 21/10 = 3. So the page count is not inclusive.